.github/workflows/ci-extension.yml

```diff
@@ -117,24 +163,40 @@ jobs:
           ${{ matrix.platform.cc }} --version
           ${{ matrix.platform.cxx }} --version
 
-      - name: Build
+      - name: Build (Linux)
+        if: startsWith(runner.os, 'Linux')
+        working-directory: CBaseNPC
+        run: |
+          mkdir build
+          cd build
+          python ../configure.py --extension-only --enable-auto-versioning --enable-optimize --sdks=${{ join(fromJSON(env.SDKS)) }} \
+            --mms-path="$GITHUB_WORKSPACE/${{ env.CACHE_PATH }}/metamod" \
+            --hl2sdk-root="$GITHUB_WORKSPACE/${{ env.CACHE_PATH }}" \
+            --sm-path="$GITHUB_WORKSPACE/${{ env.CACHE_PATH }}/sourcemod"
+          ambuild
+
+      - name: Build (Windows)
+        if: startsWith(runner.os, 'Windows')
         working-directory: CBaseNPC
         run: |
           mkdir build
           cd build
-          python ../configure.py --extension-only --enable-auto-versioning --enable-optimize --sdks=${{ join(fromJSON(env.SDKS)) }} --mms-path=${{ env.CACHE_PATH }}/metamod --hl2sdk-root=${{ env.CACHE_PATH }} --sm-path=${{ env.CACHE_PATH }}/sourcemod
+          python ../configure.py --extension-only --enable-auto-versioning --enable-optimize --sdks=${{ join(fromJSON(env.SDKS)) }} `
+            --mms-path="$env:GITHUB_WORKSPACE/${{ env.CACHE_PATH }}/metamod" `
+            --hl2sdk-root="$env:GITHUB_WORKSPACE/${{ env.CACHE_PATH }}" `
+            --sm-path="$env:GITHUB_WORKSPACE/${{ env.CACHE_PATH }}/sourcemod"
```

Comment:
Why was this split ?

Reply:
Because accessing the `$GITHUB_WORKSPACE` environment variable is slightly different on Windows (need to prefix with `$env:`). And also, `github.workspace` [isn't configured correctly in a container environment](https://github.com/actions/runner/issues/2058) as it points to a directory within the host, not in the container.

In addition, both the checkout and cache actions use `$GITHUB_WORKSPACE` [as the default location](https://github.com/actions/checkout?tab=readme-ov-file#checkout-v4) and the [base folder for relative paths](https://github.com/actions/toolkit/blob/6dd369c0e648ed58d0ead326cf2426906ea86401/packages/cache/src/internal/cacheUtils.ts#L48) respectively.

Reply:
Have you tried to force run it as a bash shell `shell: bash` ? Alternatively, what about the hack of feeding `$GITHUB_ENV` which would then make something like `${{ env.GITHUB_WORKSPACE }}` available.

Reply:
It'd be nice if we don't have to split this part, we can't help the setup part and that's to be expected. But at the minimum I'd like the compile part to be the same on both platforms, ambuild is agnostic and it'd be a shame to lose that.

If it can't be helped, then I'll take this as-is.

Reply:
I'll try running this in the bash shell for both platforms